tests/mgbench/runners.py

```diff
@@ -131,12 +132,132 @@ def execute(
             data = ret.stdout.decode("utf-8").strip().split("\n")
             if error and error[0] != "":
-                print("Reported errros from client")
-                print(error)
+                log.warning("Reported errors from client:")
+                log.warning("There is a possibility that query from: {} is not executed properly".format(file_path))
+                log.error(error)
+                log.error("Results for this query or benchmark run are probably invalid!")
             data = [x for x in data if not x.startswith("[")]
             return list(map(json.loads, data))
 
 
+class BoltClientDocker(BaseClient):
+    def __init__(self, benchmark_context: BenchmarkContext):
+        self._client_binary = benchmark_context.client_binary
+        self._directory = tempfile.TemporaryDirectory(dir=benchmark_context.temporary_directory)
+        self._username = ""
+        self._password = ""
+        self._bolt_port = (
+            benchmark_context.vendor_args["bolt-port"]
+            if "bolt-port" in benchmark_context.vendor_args.keys()
+            else "7687"
+        )
+        self._container_name = "bolt_client_benchmark"
+        self._target_db_container = (
+            "memgraph_benchmark" if "memgraph" in benchmark_context.vendor_name else "neo4j_benchmark"
+        )
+
+    def _remove_container(self):
+        command = ["docker", "rm", "-f", self._container_name]
+        self._run_command(command)
+
+    def _create_container(self, *args):
+        command = ["docker", "create", "--name", self._container_name, "memgraph/mgbench", *args]
+        self._run_command(command)
+
+    def _get_target_ip(self):
+        command = ["docker", "inspect", "--format", "{{ .NetworkSettings.IPAddress }}", self._target_db_container]
+        ret = self._run_command(command)
+        ip_address = ret.stdout.strip("\n")
+        return ip_address
+
+    def _get_logs(self):
+        command = [
+            "docker",
+            "logs",
+            self._container_name,
+        ]
+        ret = self._run_command(command)
+        return ret
+
+    def _get_args(self, **kwargs):
+        return _convert_args_to_flags(**kwargs)
+
+    def execute(
+        self,
+        queries=None,
+        file_path=None,
+        num_workers=1,
+        max_retries: int = 50,
+        validation: bool = False,
+        time_dependent_execution: int = 0,
+    ):
+        if (queries is None and file_path is None) or (queries is not None and file_path is not None):
+            raise ValueError("Either queries or input_path must be specified!")
+
+        queries_json = False
+        if queries is not None:
+            queries_json = True
+            file_path = os.path.join(self._directory.name, "queries.json")
+            with open(file_path, "w") as f:
+                for query in queries:
+                    json.dump(query, f)
+                    f.write("\n")
```

Comment:
Rename to something more specific because `queries_json` is actually `query_params_jsonl`